sat/loose_version.py

```diff
@@ -0,0 +1,80 @@
+#
+# MIT License
+#
+# (C) Copyright 2019-2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Class for representing a semver version
+"""
+from semver import Version
+
+
+class LooseVersion:
+    """A LooseVersion representing a product_version in the system"""
```

Comment:
If you want to document the `comparison_version` attribute, you could do so here. I might also generalize the description and describe the meaning of "loose" in `LooseVersion`. E.g.:

```suggestion
    """A LooseVersion representing a version that may be semver or may not.

    This class is used to compare versions that may or may not comply with the
    semver format. If the version does not comply with the semver format, then
    the MAX_VERSION is used as the comparison version, which results in the
    version being considered greater than any other version.

    Args:
        version_str (str): The string representation of the version.

    Attributes:
        version_str (str): The string representation of the version.
        comparison_version (semver.version.Version): The semver.version.Version
            object of either the version_str or the MAX_VERSION if semver fails to
            parse the version
    """
```

Per [this guide](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html), it says that we should then document the args to `__init__` there too, so then we'd drop the `__init__` docstring I suppose. I don't think it's a big deal either way. We probably have not been good about adding attributes to class docstrings throughout this code.